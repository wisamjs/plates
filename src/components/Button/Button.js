import React from "react";
import { TouchableOpacity, Text } from "react-native";
import colors from "../../utils/colors";

const defaultStyles = {
  backgroundColor: colors.tundora,
  fontColor: colors.white,
  paddingTop: 20,
  paddingBottom: 20,
  paddingLeft: 45,
  paddingRight: 45,
  width: "100%"
};

const inactiveStyles = {
  backgroundColor: colors.bermuda
};
const textStyles = {
  color: colors.white,
  fontWeight: "bold",
  textAlign: "center"
};

const getStyles = (styles, disabled) => {
  const allStylesDisabled = { ...defaultStyles, ...styles, ...inactiveStyles };
  const allStyles = { ...defaultStyles, ...styles };
  return disabled ? allStylesDisabled : allStyles;
};

export default function Button(props) {
  const allStyles = getStyles(props.style, props.disabled);

  return (
    <TouchableOpacity style={allStyles} {...props}>
      <Text style={textStyles}>{props.children}</Text>
    </TouchableOpacity>
  );
}
