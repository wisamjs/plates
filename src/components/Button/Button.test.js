import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  describe('snapshot', () => {
    it('renders a disabled button', () => {
      const tree = renderer
        .create(<Button disabled={true}>Text</Button>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a regular button', () => {
      const tree = renderer.create(<Button>Text</Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
