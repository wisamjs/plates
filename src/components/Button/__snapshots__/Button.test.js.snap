// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot renders a disabled button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#78DCBB",
      "fontColor": "#FFF",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 45,
      "paddingRight": 45,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Text
  </Text>
</View>
`;

exports[`Button snapshot renders a regular button 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#1EC58E",
      "fontColor": "#FFF",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 45,
      "paddingRight": 45,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Text
  </Text>
</View>
`;

exports[`Button snapshot renders active button with styles correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#1EC58E",
      "fontColor": "#FFF",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 45,
      "paddingRight": 45,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Text
  </Text>
</View>
`;
